packages/extension-base/src/background/types.ts

```diff
@@ -141,7 +141,7 @@ export type MessageTypesWithNullRequest = NullKeys<RequestTypes>
 export interface TransportRequestMessage<TMessageType extends MessageTypes> {
   id: string;
   message: TMessageType;
-  origin: 'page' | 'extension';
+  origin: string;
```

Comment:
```suggestion
  origin: `${string}${'page' | 'extension'}`;
```

... but also no need for magic, so 100% fine and correct as-is